cmd/terramate/cli/cloud.go

```diff
@@ -477,30 +461,91 @@ func (c *cli) detectCloudMetadata() {
 	}
 }
 
-func getGithubCommit(ghClient *github.Client, repo string, commitName string) (*github.Commit, error) {
+func getGithubCommit(ghClient *github.Client, owner, repo, commit string) (*github.RepositoryCommit, error) {
 	ctx, cancel := context.WithTimeout(context.Background(), defaultGithubTimeout)
 	defer cancel()
 
-	commit, err := ghClient.Commit(ctx, repo, commitName)
+	rcommit, _, err := ghClient.Repositories.GetCommit(ctx, owner, repo, commit, nil)
 	if err != nil {
 		return nil, err
 	}
 
-	return commit, nil
+	return rcommit, nil
 }
 
-func getGithubPR(ghClient *github.Client, repo string, commitName string) (github.Pulls, error) {
+func listGithubPRs(ghClient *github.Client, owner, repo, commit string) ([]*github.PullRequest, error) {
 	ctx, cancel := context.WithTimeout(context.Background(), defaultGithubTimeout)
 	defer cancel()
 
-	return ghClient.PullsForCommit(ctx, repo, commitName)
+	opt := &github.ListOptions{PerPage: 100}
+
+	var allPulls []*github.PullRequest
+	for {
+		pulls, resp, err := ghClient.PullRequests.ListPullRequestsWithCommit(ctx, owner, repo, commit, opt)
+		if err != nil {
+			return nil, err
+		}
+
+		allPulls = append(allPulls, pulls...)
+		if resp.NextPage == 0 {
+			break
+		}
+		opt.Page = resp.NextPage
+	}
```

Comment:
I wonder if this needs pagination if we only use the first result... wdyt?

Reply:
Good point. Basically they are obtained just to print them for debug purposes, then everything is discarded but the first one. Maybe this function should just get the latest pull request in the first place.